vpr/src/analytical_place/analytical_solver.h

```diff
@@ -301,6 +310,213 @@ class QPHybridSolver : public AnalyticalSolver {
      *                      this object.
      */
     void solve(unsigned iteration, PartialPlacement& p_placement) final;
+
+    /**
+     * @brief Print statistics of the solver.
+     */
+    void print_statistics() final;
+};
+
+/**
+ * @brief An Analytical Solver which tries to minimize the linear HPWL objective:
+ *          SUM((xmax - xmin) + (ymax - ymin)) over all nets.
+ *
+ * This is implemented using the Bound2Bound method, which iteratively sets up a
+ * linear system of equations (similar to the QP Hybrid approach above) which
+ * solves a quadratic objective function. For a net model, each block connects
+ * to the current bounding blocks in the given dimension and the weight of this
+ * connection is inversly proportional to the distance of the block to the bound.
+ * After minimizing this system, the bounds are likely to change; so the system
+ * needs to be reconstructed and solved iteratively.
+ *
+ * This technique was proposed in Kraftwerk2, where they proved that the B2B Net
+ * Model will, in theory, converge on the linear HPWL solution.
+ *          https://doi.org/10.1109/TCAD.2008.925783
+ */
```

Comment:
I think I previously asked you to add some comments on the B2B model and how the matrices are constructed. Given the explanation here, feel free to ignore that request.